sdcm/results_analyze/__init__.py

```diff
@@ -604,8 +602,23 @@ def check_regression(self, test_id, is_gce=False, email_subject_postfix=None, us
             subject = f'{subject} (ebs volume type {ebs_type})'
         if email_subject_postfix:
             subject = f'{subject} {email_subject_postfix}'
-        html = self.render_to_html(results)
-        self.send_email(subject, html)
+
+        email_data = {'email_body': results,
+                      'attachments': (),
+                      'template': self._email_template_fp}
+
+        file_path = 'email_data.json'
+        if os.path.exists(file_path):
+            with open(file_path, 'r') as file:
+                data = file.read().strip()
+                file_content = json.loads(data or '{}')
+        else:
+            file_content = dict()
+
+        file_content[subject] = email_data.copy()
+
+        with open(file_path, 'w') as file:
+            json.dump(file_content, file)
```

Comment:
This code repeated in each check_regression method. May it is better to create new method in Base result analyzer with nae save_results_for_email or prepare_email

Reply:
dope